src/box/lua/config/init.lua

```diff
@@ -481,6 +500,55 @@ function methods.info(self)
     }
 end
 
+-- Cluster configuration.
+--
+-- It is given as is after merging from all the configuration
+-- sources. Default values are NOT applied. Variables are NOT
+-- substituted.
+--
+-- Use config:get() to receive an instance config for a particular
+-- instance with applied defaults and substituted variables.
+function methods.cconfig(self)
+    selfcheck(self, 'cconfig')
+    initcheck(self, 'cconfig', 'cluster')
+
+    return self._configdata_applied:cconfig()
+end
+
+-- List all instances of the cluster.
+--
+-- Returns a table of the following format.
+--
+-- {
+--     [<instance_name>] = {
+--         instance_name = <...>,
+--         replicaset_name = <...>,
+--         group_name = <...>,
+--     },
+--     <...>
+-- }
```

Comment:
@locker I have changed the shape of the return value a bit comparing to the workaround I've provided in #9680. The keys are instance names now.

It feels natural, because the order is anyway unspecified. It also allows easier access to the instance name.

Before (however, it works now too):

```lua
for _, def in pairs(config:instances())
    action(def.instance_name)
end
```

Now:

```lua
for instance_name in pairs(config:instances())
    action(instance_name)
end
```

If you're using `ipairs()` for traversing over `config:instance()`, it doesn't work after the change.

What do you think about this change?

Reply:
This change looks good to me (although I do use `ipairs`).